pkgs/top-level/all-packages.nix

```diff
@@ -5298,7 +5298,7 @@ in
   opentracing-cpp = callPackage ../development/libraries/opentracing-cpp { };
 
   openvswitch = callPackage ../os-specific/linux/openvswitch {
-    openssl = openssl_1_0_2;
+    openssl = __splicedPackages.openssl_1_0_2;
```

Comment:
/cc @ericson2314

Reply:
Is `__splicedPackages` targetPackages?

Reply:
Or in other words, what does this change fixes?

Reply:
`openssl_1_0_2` is just a derivation, whereas `__splicedPackages.openssl_1_0_2` is augmented with attributes referring to that derivation for other `(host, target)` pairs. For example, `__splicedPackages.openssl_1_0_2.nativeDrv` refers to `buildPackages.openssl_1_0_2`.

See this comment for more detail: https://github.com/NixOS/nixpkgs/issues/68967#issuecomment-532432617

Reply:
`openvswitch/default.nix` refers to `openssl` in both `.nativeBuildInputs` and `.buildInputs`. In the case of cross-compilation, `mkDerivation` accesses spliced attributes for elements of these lists. So, the `openssl` argument to `openvswitch/default.nix` must be spliced.

`callPackage` automatically uses spliced packages. However, the packages in scope in `top-level/all-packages.nix` are not spliced. So, to pass in a spliced `openssl_1_0_2`, we must explicitly refer to `__splicedPackages.openssl_1_0_2`.

This may not be the best solution to cross-compiling this particular package, but similar situations appear throughout Nixpkgs. I think it's worth considering how we should be handling them.

Reply:
Ok. @Ericson2314 seems to prefer `buildPackages` though.

Reply:
I agree with them. This case is a bit more complicated because `openvswitch` really takes two distinct `openssl` derivations as parameters, one for the build and one to link against, which are different in the case of cross compilation. They suggest what seems like the most sensible solution right now: https://github.com/NixOS/nixpkgs/pull/68900#discussion_r436745733. That is, to explicitly allow the overriding of both derivations separately.